packages/cli/test/generators/app.test.ts

```diff
@@ -0,0 +1,237 @@
+
+import AppGenerator from '../../src/generators/app'
+
+// Spies process to avoid trying to chdir to a non existing folder
+jest.spyOn(process, 'chdir').mockImplementation(() => true)
+// Sypies the console log to not console on tests
+jest.spyOn(global.console, 'log').mockImplementation(() => undefined)
+// Mocks the log output
+jest.mock(
+  '@blitzjs/server',
+  jest.fn(() => {
+    return {
+      log: {
+        withBranded: jest.fn(),
+        spinner: jest.fn().mockImplementation(() => {
+          return {
+            start: jest.fn().mockImplementation(() => ({succeed: jest.fn()})),
+          }
+        }),
+      },
+    }
+  }),
+)
+
+// Mocks spawn
+jest.mock('cross-spawn')
+
+// Mocks fs-extra used by both AppGenerator and Generator base class
+jest.mock(
+  'fs-extra',
+  jest.fn(() => {
+    return {
+      readJSONSync: jest.fn().mockImplementation(() => ({
+        dependencies: {
+          react: '^16.8.0',
+        },
+        devDependencies: {
+          debug: '^4.1.1',
+        },
+      })),
+      ensureDir: jest.fn(),
+      existsSync: jest.fn(),
+      writeJson: jest.fn(),
+    }
+  }),
+)
+
+jest.mock(
+  'mem-fs-editor',
+  jest.fn(() => {
+    return {
+      create: jest.fn().mockImplementation(() => {
+        return {
+          move: jest.fn(),
+          commit: (_: any, callback: any) => callback(),
+          copy: jest.fn(),
+        }
+      }),
+    }
+  }),
+)
+
+describe('AppGenerator', () => {
+  beforeEach(() => {
+    // @ts-ignore TS avoid typing the mocked spawn
+    spawn.sync = jest.fn().mockImplementation(() => ({status: 0}))
+    jest.clearAllMocks()
+  })
+
+  const generator = new AppGenerator({
+    sourceRoot: '../../templates/app',
+    destinationRoot: '/new-app',
+    appName: 'new-app',
+    dryRun: false,
+    useTs: true,
+    yarn: true,
+    version: '1.0',
+  })
+
+  it('calls commands in correct order', async () => {
+    const order: Array<string> = []
+    // @ts-ignore TS avoid typing the mocked spawn
+    spawn.sync = jest.fn().mockImplementation((cli: string, options: Array<string>) => {
+      const fullCommand = [cli, ...options].join(' ')
+      order.push(fullCommand)
+      return {status: 0}
+    })
+
+    await generator.run()
+    expect(order).toEqual([
+      'git init',
+      'yarn install',
+      'yarn run prettier --loglevel silent --write .',
+      'git add .',
+      'git commit -m New baby Blitz app!',
+    ])
+  })
+
+  describe('when install command fails', () => {
+    it('throws error', async () => {
+      // @ts-ignore TS avoid typing the mocked spawn
+      spawn.sync = jest.fn().mockImplementation((cli: string, options: Array<string>) => {
+        if (cli == 'yarn' && options[0] == 'install') {
+          return {status: 1}
+        } else {
+          return {status: 0}
+        }
+      })
+      try {
+        await generator.run()
+      } catch (e) {
+        expect(e).toEqual(new Error('Failed to yarn install'))
+      }
```

Comment:
Tried hard to make this work with `toThrow`, but seems to be a known issue.
https://github.com/facebook/jest/issues/1700